extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getInstartLogicBootstrap.mustache

```diff
@@ -1,11 +1,19 @@
 <script type="text/javascript">
 	require([
+		'wikia.document',
 		'ext.wikia.adEngine.adContext'
-	], function (adContext) {
+	], function (doc, adContext) {
 		var context = adContext.getContext();
 
+
 		if (context.opts.instartLogicRecovery) {
-			{{{code}}}
+			if (context.opts.babRecovery) {
+				document.addEventListener('bab.blocking', function () {
+					{{{code}}}
```

Comment:
this code block is pretty huge. Can we somehow avoid loading it two times?

Reply:
Only one working solution I found out: applied.